### ISSUE TITLE
Rename "Resolution" and "Test Resolution" to clearer terms

### ISSUE THREAD

**Issue by Sosasees** [godotengine/godot-proposals#2822]
### Describe the project you are working on

I am working on a Calculator for learning purposes

### Describe the problem or limitation you are having in your project

Because these 2 settings are named "Resolution" and "Test Resolution", I expected the "Test Resolution" to only apply when the game gets tested in the Editor, and then revert back to the base "Resolution" in the exported version of the game.

Instead, something much better happens:
I first heard of this fact, and then confirmed it for myself:

The "Resolution" is still the base resolution of the game, and then it gets resized to the "Test Resolution", reguardless of it runs in the editor or was exported.

Just as expected, running a game
with a Resolution of 300×300 and Test Resolution of 600×600
is different from running it at an identical Resolution and Test Resolution of 600×600,
in that with the former option, the window is still the same size but the GUI elements are twice as large.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

In Website Development, the same concepts are named
- "Logical Resolution" instead of "Resolution"
- "Physical Resolution" instead of "Test Resolution"
These alternative terms describe much better what these settings actually do, so that I think they should be the official terms.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

- Rename "Resolution" to "Logical Resolution"
- Rename "Test Resolution" to "Physical Resolution"
- Rename the GDScript references of these project settings to ``logical_resolution`` and ``physical_resolution`` respectively

### If this enhancement will not be used often, can it be worked around with a few lines of script?

No

### Is there a reason why this should be core and not an add-on in the asset library?

This is a clarity change for existing core functionality.

**Comment by LikeLakers2** [comment]
While I understand the reasoning for the suggested rename, I fear that the meanings of "Logical" and "Physical" may get lost on new people.

Perhaps you could use "Resolution (when exported)" and "Resolution (when running from editor)", or something of the sort?

**Comment by Sosasees** [comment]
How about "Base Resolution" amd "Actual Resolution", then?

**Comment by LikeLakers2** [comment]
@Sosasees To be honest, I find that even more confusing than "Logical Resolution" and "Physical Resolution". We should ensure that new users are able to tell which is which at a glance, and which one applies to which situation, and I don't think "Base Resolution" or "Actual Resolution" handle either of those problems.

**Comment by Sosasees** [comment]
So, we should just leave the terms as the current
"Resolution" and "Test Resolution"?

**Comment by LikeLakers2** [comment]
Oh, I'm not suggesting that. You made this issue because you thought they could be clearer -- so I think it makes sense to think up names that we think are clearer.

But the issue I'm pointing out is that the suggestions you've made thus far don't seem friendly to newcomers. They sound like terms that newcomers will have to Google, despite the topic being as simple as window resolution, which is why I'm not a fan of them.

**Comment by Sosasees** [comment]
If anyone has other suggestions for better names, or wants to share their opinion on name suggestions made before in this issue, please reply.
This issue does not yet have enough opinions to be solved.

**Comment by KoBeWi** [comment]
Related: https://github.com/godotengine/godot/pull/47522

**Comment by Sosasees** [comment]
@KoBeWi This sounds good.
In my opinion, these are the best name suggestions (so far):
- ``base_width`` or ``base_resolution.x``, instead of ``resolution.x``
- ``base_height`` or ``base_resolution.y``, instead of ``resolution.y``
- ``window_width`` and ``window_height``, or``window_size``, instead of ``test_width`` and ``test_height``

**Comment by mrjustaguy** [comment]
See https://github.com/godotengine/godot/issues/16863

**Comment by Sosasees** [comment]
@mrjustaguy I have just looked a bit through this issue, and:
yes, to me the names we have come up with in this conversation
Are clearer than the previously clearest names from https://github.com/godotengine/godot/issues/16863 :

Now I want to hear other people's reasons for agreeing or disagreeing with this statement.

**Comment by mrjustaguy** [comment]
I'd say Window Size and Render Size are clearer, as most games that have the 2, usually follow this convention.

**Comment by Sosasees** [comment]
OK, I'll have to re-check what these project settings actually change.

**Comment by mrjustaguy** [comment]
Window Size is how large the window is on screen, and is normally Full screen & Native Res, while Render size is the size of the Viewport that is rendered, and then up/down-scaled to the Window size.. 
Example:
1920x1080 window size (on a 1080p display)
960x540 render size
Result:
Full screen, Half Resolution Renderings upscaled to 1080p

Now I don't know how exactly it works in Godot with UI, but normally UI is changed with Window Size, with the 3D being changed with Render Size. So UI Changes with Window Size changes, and 3D changes with Render size, in theory.

**Comment by LikeLakers2** [comment]
@Sosasees I feel silly. I reread your original post, and apparently I got caught in the same trap you did -- thinking that "Test Resolution" meant the resolution it would be at while testing from the editor. My bad, sorry!

Since I realize my error now, I want to say that I now agree with your suggestion for "Base Resolution" and "Actual Resolution". I don't think those are confusing anymore, although I will suggest one better: "Render Resolution" and "Window Resolution" (respectively).

**Comment by Sosasees** [comment]
After playing around with 'window size' and 'test window size' and 'stretch modes' in a 2D scene with only icon.png, i think
„Yes: 'base resolution' and 'window resolution' are the correct names”

Although I think that 'window resolution' and 'render resolution' are a little misleading in comparison,
if most people understand these terms better, these should be the terms.

I recommend playing around with these settings as well to see which terms are more accurate to you.

**Comment by LikeLakers2** [comment]
> Although I think that 'window resolution' and 'render resolution' are a little misleading in comparison,

I'm a little confused how it might be misleading. Let me explain why I named them that.

From how I understood the explanation in your original post, one setting (which I call "Render Resolution") is the resolution that the game is rendered at, and the other (which I call "Window Resolution") is the resolution of the game window (which the game's rendered image gets resized to fit into).

Am I misunderstanding something about this? Or is there something else that makes it misleading to you?

**Comment by Sosasees** [comment]
Oops, i thought that 'window resolution' and 'render resolution' were in the wrong order.
In this order ('render resolution' first, 'window resolution' second), it is correct.

Although the term 'render resolution' is only 100% correct if the stretch mode is set to 'viewport'.
Because if the stretch mode is set to '2d' or 'disabled', the 'render resolution' remains the base resolution while the Actual render resolution becomes as large as the window size.

**Comment by ryliejamesthomas** [comment]
See also:

- Similar issue: https://github.com/godotengine/godot/issues/35041
- Various things to be renamed: https://github.com/godotengine/godot/issues/16863
- Pull request: https://github.com/godotengine/godot/pull/47522

**Comment by madmiraal** [comment]
This is a duplicate of #3769 that was closed in https://github.com/godotengine/godot/pull/47522, but with the suggestion:
> But feel free to continue discussion if there are ideas to improve this further.

That been said, the title of this proposal is confusing. As discussed in #3769, neither setting refers to resolution. Godot does not define the resolution. The device the game is played on defines the resolution. This is either fixed to the full screen resolution (on mobile devices) or defined by the window size (on desktop devices). The user specifies the `Viewport` size, which is converted to the actual screen or window size ("resolution"); using the `stretch` modes.

In https://github.com/godotengine/godot/pull/47522, these settings were changed to `viewport_width` and `viewport_height`, and `window_width_override` and `window_height_override`. The title of this proposal needs to reflect that or be closed.

**Comment by Sosasees** [comment]
Yes.
This is much better explained in [godotengine/godot#47522](https://github.com/godotengine/godot/pull/47522) than here. 
Only after reading that issue, i realized that my proposed replacemeny terms here are not that good, while the terms proposed there are good.
Thanks for pointing me to that issue.
